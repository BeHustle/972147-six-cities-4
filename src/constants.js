const CARD_TYPES = [
  `Apartment`,
  `Private room`
];

const Cities = {
  AMSTERDAM: [52.38333, 4.9]
};

const MAP_ZOOM = 12;

const ICON_SIZE = [30, 30];

const ICON_PATH = `img/pin.svg`;

export {CARD_TYPES, Cities, MAP_ZOOM, ICON_SIZE, ICON_PATH};
