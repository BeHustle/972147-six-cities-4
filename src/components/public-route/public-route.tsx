import React from 'react';
import {Route, Redirect} from 'react-router-dom';
import {connect} from "react-redux";
import AppRoute from '../../routes';
import {AuthStatus} from '../../constants';
import {getAuthStatus} from '../../reducer/user/user.selectors';
import PropTypes from 'prop-types';


const PublicRoute = ({children, path, exact, authStatus}) =>
  <Route path={path} exact={exact}>
    {authStatus === AuthStatus.NO_AUTH
      ? children
      : <Redirect to={AppRoute.MAIN}/>}
  </Route>;

PublicRoute.propTypes = {
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  authStatus: PropTypes.oneOf(Object.values(AuthStatus)),
  children: PropTypes.object
};

const mapStateToProps = (state) => ({
  authStatus: getAuthStatus(state),
});


export {PublicRoute};
export default connect(mapStateToProps)(PublicRoute);
