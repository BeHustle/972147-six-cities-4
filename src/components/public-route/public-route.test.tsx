import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {PublicRoute} from "./public-route";
import {AuthStatus} from '../../constants';
import AppRoute from '../../routes';
import {Router} from 'react-router-dom';
import {history} from '../../history';

const MockComponent = () => <div />;

it(`Render public route`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <PublicRoute
            authStatus={AuthStatus.NO_AUTH}
            path={AppRoute.LOGIN}
            exact={true}>
            <MockComponent />
          </PublicRoute>
        </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
