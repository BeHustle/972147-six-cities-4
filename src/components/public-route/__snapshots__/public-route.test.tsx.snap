// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render public route 1`] = `null`;
